grammarinator/resources/codegen/GeneratorTemplate.py.jinja

```diff
@@ -47,25 +47,26 @@ if self._max_depth >= {{ 0 if node.min == 1 else node.min_depth }}:
 
 
 {% macro processAlternationNode(node) %}
-choice = self._model.choice(current, {{ node.idx }}, [0 if [{{ node.min_depth | join(', ') }}][i] > self._max_depth else w for i, w in enumerate([{{ node.conditions | join(', ') }}])])
-{% set simple_lits, simple_rules = node.simple_alternatives() %}
-{% if simple_lits and simple_rules %}
-src = [{% for lit in simple_lits %}{% if lit is not none %}'{{ lit }}'{% else %}None{% endif %}{% if not loop.last %}, {% endif %}{% endfor %}][choice]
-rule = [{% for rule in simple_rules %}{% if rule is not none %}self.{{ rule }}{% else %}None{% endif %}{% if not loop.last %}, {% endif %}{% endfor %}][choice]
-if src is not None:
-    UnlexerRule(src=src, parent=current)
-else:
-    rule(parent=current)
-{% elif simple_lits %}
-UnlexerRule(src=[{% for lit in simple_lits %}'{{ lit }}'{% if not loop.last %}, {% endif %}{% endfor %}][choice], parent=current)
-{% elif simple_rules %}
-[{% for rule in simple_rules %}self.{{ rule }}{% if not loop.last %}, {% endif %}{% endfor %}][choice](parent=current)
-{% else %}
-{% for child in node.out_neighbours %}
-{{ 'if' if loop.index0 == 0 else 'elif' }} choice == {{ loop.index0 }}:
-    {{ processNode(child) | indent -}}
-{% endfor %}
-{% endif %}
+with AlternationContext(self, [{{ node.min_depth | join(', ') }}], [{{ node.conditions | join(', ') }}]) as alt{{ node.idx }}:
+    choice{{ node.idx }} = self._model.choice(current, {{ node.idx }}, alt{{ node.idx }})
```

Comment:
It would be a little bit clearer IMO if `alt{{ node.idx }}` would be renamed to `weights{{ node.idx }}`. The rationale is that although the "result" of AlternationContext is stored in `alt{{ node.idx }}`, but that result is not an alternation node, or context, or whatever. It is a list of weights that will be passed to `choice`.